bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java

```diff
@@ -0,0 +1,692 @@
+                break;
+            case "REWIND":
+                updateState(CHANNEL_PLAYER, RewindFastforwardType.REWIND);
+                break;
+        }
+        if (state != null) {
+            logger.debug("{}: Setting Player state to {}", thingId, state);
+            updateState(CHANNEL_PLAYER, state);
+        }
+    }
+
+    /**
+     * Connect to the receiver
+     *
+     * @throws MagentaTVException something failed
+     */
+    protected void connectReceiver() throws MagentaTVException {
+        if (control.checkDev()) {
+            updateThingProperties();
+            handlerFactory.registerDevice(config.getUDN(), config.getTerminalID(), config.getIpAddress(), this);
+            control.subscribeEventChannel();
+            control.sendPairingRequest();
+
+            // check for pairing timeout
+            final int iRefresh = ++idRefresh;
+            pairingWatchdogJob = scheduler.schedule(() -> {
+                if (iRefresh == idRefresh) { // Make a best effort to not run multiple deferred refresh
+                    if (config.getVerificationCode().isEmpty()) {
+                        setOnlineState(ThingStatus.OFFLINE, "Timeout on pairing request!");
+                    }
+                }
+            }, 15, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * If userID is empty and credentials are given the Telekom OAuth service is
+     * used to query the userID
+     *
+     * @throws MagentaTVException
+     */
+    private void authenticateUser() throws MagentaTVException {
+        String userId = config.getUserID();
+        if (userId.isEmpty()) {
+            // run OAuth authentication, this finally provides the userID
+            userId = control.authenticateUser(config.getAccountName(), config.getAccountPassword());
+
+            // Update thing configuration (persistent) - remove credentials, add userID
+            Configuration configuration = this.getConfig();
+            configuration.remove(PROPERTY_ACCT_NAME);
+            configuration.remove(PROPERTY_ACCT_PWD);
+            configuration.remove(PROPERTY_USERID);
+            configuration.put(PROPERTY_ACCT_NAME, "");
+            configuration.put(PROPERTY_ACCT_PWD, "");
+            configuration.put(PROPERTY_USERID, userId);
+            this.updateConfiguration(configuration);
+            config.setAccountName("");
+            config.setAccountPassword("");
+        } else {
+            logger.debug("{}: Skip OAuth, use existing userID {}", thingId, config.getUserID());
+        }
+        if (!userId.isEmpty()) {
+            config.setUserID(userId);
+        } else {
+            logger.warn("{}: Unable to obtain userId from OAuth", thingId);
+        }
+    }
+
+    /**
+     * Update thing status
+     *
+     * @param mode new thing status
+     * @return ON = power on, OFF=power off
+     */
+    public void setOnlineState(ThingStatus newStatus, String errorMessage) {
```

Comment:
Shouldn't this be called `setOnlineStatus`?

Reply:
changed